File: Extractor/Gradle/plugin/core/src/main/kotlin/net/meilcli/bibliothekar/extractor/plugin/core/entities/Pom.kt
package net.meilcli.bibliothekar.extractor.plugin.core.entities

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Pom(
    @SerialName("group")
    val group: String,

    @SerialName("artifact")
    val artifact: String,

    @SerialName("version")
    val version: String,

    @SerialName("name")
    val name: String?,

    @SerialName("description")
    val description: String?,

    @SerialName("url")
    val url: String?,

    @SerialName("parent")
    val parent: PomParent?,

    @SerialName("organization")
    val organization: PomOrganization?,

    @SerialName("licenses")
    val licenses: List<PomLicense>,

    @SerialName("developers")
    val developers: List<PomDeveloper>
) {

    fun toDependency(): Dependency {
        return Dependency(
            group = group,
            name = artifact,
            version = version
        )
    }
}

File: Transformer/NodeJs/core/src/maven/pom.ts
/**
 * @see {isPom} ts-auto-guard:type-guard
 */
export interface Pom {
    group: string;
    artifact: string;
    version: string;
    name: string | null;
    description: string | null;
    url: string | null;
    parent: PomParent | null;
    organization: PomOrganization | null;
    licenses: PomLicense[] | null;
    developers: PomDeveloper[] | null;
}

/**
 * @see {isPomParent} ts-auto-guard:type-guard
 */
export interface PomParent {
    group: string;
    artifact: string;
    version: string;
}

/**
 * @see {isPomOrganization} ts-auto-guard:type-guard
 */
export interface PomOrganization {
    name: string | null;
}

/**
 * @see {isPomLicense} ts-auto-guard:type-guard
 */
export interface PomLicense {
    name: string | null;
    url: string | null;
}

/**
 * @see {isPomDeveloper} ts-auto-guard:type-guard
 */
export interface PomDeveloper {
    name: string | null;
    organization: string | null;
}
